src/aind_metadata_service/slims/ecephys/handler.py

```diff
@@ -0,0 +1,439 @@
+"""
+Module to handle fetching ephys data from slims and parsing it to a model
+"""
+
+from datetime import datetime
+from typing import List, Optional, Tuple
+
+from networkx import DiGraph, descendants
+from pydantic import BaseModel
+from slims.criteria import equals
+from slims.internal import Record
+
+from aind_metadata_service.slims.table_handler import (
+    SlimsTableHandler,
+    get_attr_or_none,
+)
+
+
+class SlimsStreamModule(BaseModel):
+    """Stream Module"""
+
+    implant_hole: Optional[int] = None
+    assembly_name: Optional[str] = None
+    probe_name: Optional[str] = None
+    primary_target_structure: Optional[str] = None
+    secondary_target_structures: Optional[list] = None
+    arc_angle: Optional[float] = None
+    module_angle: Optional[float] = None
```

Comment:
Using Decimals instead of floats might be better to avoid floating-point inaccuracies.